libpod/runtime_img.go

```diff
@@ -645,36 +645,35 @@ func (r *Runtime) PullImage(imgName string, options CopyOptions) error {
 		// could be trying to pull from registry with short name
 		pullStructs, err = getRegistriesToTry(imgName, r.store, r.config.ImageDefaultTransport)
 		if err != nil {
-			return errors.Wrap(err, "error getting default registries to try")
+			return "", errors.Wrap(err, "error getting default registries to try")
 		}
 	} else {
 		pullStructs, err = r.getPullListFromRef(srcRef, imgName, sc)
 		if err != nil {
-			return errors.Wrapf(err, "error getting pullStruct info to pull image %q", imgName)
+			return "", errors.Wrapf(err, "error getting pullStruct info to pull image %q", imgName)
 		}
 	}
-
 	policyContext, err := getPolicyContext(sc)
 	if err != nil {
-		return err
+		return "", err
 	}
 	defer policyContext.Destroy()
 
 	copyOptions := common.GetCopyOptions(options.Writer, signaturePolicyPath, &options.DockerRegistryOptions, nil, options.SigningOptions, options.AuthFile)
 
 	for _, imageInfo := range pullStructs {
-		if options.Writer != nil {
+		if options.Writer != nil && imageInfo.srcRef.Transport().Name() != DockerArchive && imageInfo.srcRef.Transport().Name() != OCIArchive && imageInfo.srcRef.Transport().Name() != DirTransport {
 			io.WriteString(options.Writer, fmt.Sprintf("Trying to pull %s...\n", imageInfo.image))
 		}
 		if err = cp.Image(policyContext, imageInfo.dstRef, imageInfo.srcRef, copyOptions); err != nil {
-			if options.Writer != nil {
+			if options.Writer != nil && imageInfo.srcRef.Transport().Name() != DockerArchive && imageInfo.srcRef.Transport().Name() != OCIArchive && imageInfo.srcRef.Transport().Name() != DirTransport {
```

Comment:
Not sure why you want to block the printing of the "Failed" message for these transports.

Reply:
Should go with the transport that you want to print this line.  Docker and Atomic?

Reply:
Right, we can print the "Failed" for all transports. Fixed.

Reply:
Can you make 
```

+		if options.Writer != nil && imageInfo.srcRef.Transport().Name() != DockerArchive && imageInfo.srcRef.Transport().Name() != OCIArchive && imageInfo.srcRef.Transport().Name() != DirTransport {
```
something like
```

+		if options.Writer != nil && (imageInfo.srcRef.Transport().Name() == Docker || imageInfo.srcRef.Transport().Name() != Atomic) {
```

Reply:
fixed